scripts/update-catalog.py

```diff
@@ -0,0 +1,98 @@
+"""
+from pathlib import Path
+import subprocess
+import json
+import argparse
+import yaml
+
+repository = Path().absolute().name
+application = repository.replace("konflux-", "").replace("opentelemetry", "otel")
+if not repository.startswith("konflux-"):
+    raise Exception("This script must be run from a Konflux data repository.")
+
+def get_bundle_pullspec(args):
+    p = subprocess.run(["kubectl", "get", "snapshot", args.snapshot, "-o", "json"], capture_output=True, text=True, check=True)
+    snapshot = json.loads(p.stdout)
+
+    for component in snapshot["spec"]["components"]:
+        if component["name"] == f"{application}-bundle":
+            return component["containerImage"]
+    return None
+
+def update_catalog_template(bundle_pullspec):
+    with open("bundle-patch/patch_csv.yaml", "r") as f:
+        bundle_patch = yaml.safe_load(f)
+
+    with open("catalog/catalog-template.yaml", "r") as f:
+        catalog_template = yaml.safe_load(f)
+
+    catalog_template["entries"][1]["entries"].append({
+        "name": bundle_patch["metadata"]["name"],
+        "replaces": bundle_patch["spec"]["replaces"],
+        "skipRange": bundle_patch["metadata"]["extra_annotations"]["olm.skipRange"]
+    })
+    catalog_template["entries"].append({
+        "image": bundle_pullspec,
+        "schema": "olm.bundle",
+    })
+
+    with open("catalog/catalog-template.yaml", "w") as f:
+        content = "---\n" + yaml.dump(catalog_template)
+        f.write(content)
+
+def update_catalog():
+    if application == "jaeger":
+        subprocess.run("""
+        opm alpha render-template basic --output yaml catalog/catalog-template.yaml > catalog/jaeger-product/catalog.yaml &&
+        opm alpha render-template basic --output yaml --migrate-level bundle-object-to-csv-metadata catalog/catalog-template.yaml > catalog/jaeger-product-4.17/catalog.yaml &&
+        sed -i 's#quay.io/redhat-user-workloads/rhosdt-tenant/jaeger/jaeger-bundle#registry.redhat.io/rhosdt/jaeger-operator-bundle#g' catalog/jaeger-product/catalog.yaml &&
+        sed -i 's#quay.io/redhat-user-workloads/rhosdt-tenant/jaeger/jaeger-bundle#registry.redhat.io/rhosdt/jaeger-operator-bundle#g' catalog/jaeger-product-4.17/catalog.yaml &&
+        opm validate catalog/jaeger-product &&
+        opm validate catalog/jaeger-product-4.17
+        """, shell=True, check=True)
+    elif application == "otel":
+        subprocess.run("""
+        opm alpha render-template basic --output yaml catalog/catalog-template.yaml > catalog/opentelemetry-product/catalog.yaml &&
+        opm alpha render-template basic --output yaml --migrate-level bundle-object-to-csv-metadata catalog/catalog-template.yaml > catalog/opentelemetry-product-4.17/catalog.yaml &&
+        sed -i 's#quay.io/redhat-user-workloads/rhosdt-tenant/otel/opentelemetry-bundle#registry.redhat.io/rhosdt/opentelemetry-operator-bundle#g' catalog/opentelemetry-product/catalog.yaml &&
```

Comment:
`rhosdt-tenant/otel` we could rename image location to `/opentelemetry` if it would help to simplify this script - e.g. extract some parts to variables and have a single procedure.

Reply:
Yep, we could. Not sure how difficult it is to rename the images though, probably we would need assistance from the release team. In my opinion it's ok to keep it simple and stupid as it is now, unless we get a lot more products in the future.